.github/actions/deploy-cloud-run/action.yml

```diff
@@ -0,0 +1,81 @@
+name: Deploy Cloud Run service
+description: Create a new revision and rollback if health check fails
+inputs:
+  region:
+    description: Region of the Cloud Run service
+    required: true
+  image:
+    description: Image name
+    required: true
+  service:
+    description: Name of the Cloud Run service
+    required: true
+  health_check_url:
+    description: Health check URL. Multiple URLs can be separated by line breaks. If not provided, health check will be skipped.
+
+runs:
+  using: composite
+  steps:
+    - name: Get current revision
+      id: get_revision
+      shell: bash
+      run: |
+        REV=$( \
+          gcloud run services describe "${{ inputs.service }}" \
+            --region "${{ inputs.region }}" \
+            --format="value(status.latestReadyRevisionName)" \
+        )
+        echo "Current revision: $REV"
+        echo "revision=$REV" >> "$GITHUB_OUTPUT"
+
+    - name: Deploy
+      shell: bash
+      run: |
+        gcloud run deploy "${{ inputs.service }}" \
+          --image "${{ inputs.image }}" \
+          --region "${{ inputs.region }}" \
+          --platform managed \
+          --quiet
```

Comment:
We could utilize this [GH actions](https://github.com/google-github-actions/deploy-cloudrun)